gpjax/dataset.py

```diff
@@ -60,14 +85,20 @@ def __add__(self, other: "Dataset") -> "Dataset":
         r"""Combine two datasets. Right hand dataset is stacked beneath the left."""
         X = None
         y = None
+        mask = None
 
         if self.X is not None and other.X is not None:
             X = jnp.concatenate((self.X, other.X))
 
         if self.y is not None and other.y is not None:
             y = jnp.concatenate((self.y, other.y))
 
-        return Dataset(X=X, y=y)
+        if (sm := self.mask is not None) | (om := other.mask is not None):
```

Comment:
Stylistic, but elsewhere in the package we use `or` not `|`. Could we be consistent here please unless there's a performance benefit? Above discussion on `:=` holds true here too.

Reply:
"or" and "|" are not the same. If there are two conditions:
`cond1 or cond2`: cond2 does NOT evaluate if cond1 is already True
`cond1 | cond2`: both conditions are evaluated, and then True is returned is one of them is True (which is what we want here)